x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/take_action/index.tsx

```diff
@@ -210,18 +211,24 @@ const TakeActionComponent: React.FC<Props> = ({
 
   const isAgentBuilderEnabled = useIsExperimentalFeatureEnabled('agentBuilderEnabled');
   const attackDiscovery = attackDiscoveries.length === 1 ? attackDiscoveries[0] : null;
-  const { openAgentBuilderFlyout } = useAgentBuilderAttachment({
-    attachmentType: SecurityAgentBuilderAttachments.alert,
-    attachmentData: {
-      alert: attackDiscovery
-        ? getAttackDiscoveryMarkdown({
-            attackDiscovery,
-            replacements,
-          })
-        : '',
-    },
-    attachmentPrompt: i18n.VIEW_IN_AGENT_BUILDER,
-  });
+  const alertAttachment = useMemo(
```

Comment:
nit: I see we build `alertAttachment` based on attack discovery document and replacements in a few places. Maybe we could encapsulate this logic in a hook?